packages/block-library/src/search/index.php

```diff
@@ -188,25 +213,31 @@ function styles_for_block_core_search( $attributes ) {
 	$has_border_radius = ! empty( $attributes['style']['border']['radius'] );
 
 	if ( $has_border_radius ) {
-		// Shared style for button and input radius values.
-		$border_radius   = $attributes['style']['border']['radius'];
-		$border_radius   = is_numeric( $border_radius ) ? $border_radius . 'px' : $border_radius;
-		$shared_styles[] = sprintf( 'border-radius: %s;', esc_attr( $border_radius ) );
-
-		// Apply wrapper border radius if button placed inside.
+		$border_radius = $attributes['style']['border']['radius'];
 		$button_inside = ! empty( $attributes['buttonPosition'] ) &&
 			'button-inside' === $attributes['buttonPosition'];
 
-		if ( $button_inside ) {
-			// We adjust the border radius value for the outer wrapper element
-			// to make it visually consistent with the radius applied to inner
-			// elements. calc() is used to support non-pixel CSS units.
-			$default_padding  = '4px';
-			$wrapper_styles[] = sprintf(
-				'border-radius: calc(%s + %s);',
-				esc_attr( $border_radius ),
-				esc_attr( $default_padding )
-			);
+		if ( is_array( $border_radius ) ) {
+			// Add shared styles for individual border radii for input & button.
+			$shared_styles[] = sprintf( 'border-top-left-radius: %s;', esc_attr( $border_radius['topLeft'] ) );
```

Comment:
Should these four lines be guarded behind `! empty( $border_radius['topLeft'] ) || 0 === $border_radius['topLeft']`  checks for each of the sides? Otherwise empty values for each side will result in an empty CSS value being added to the inline style (not an issue visually for the block, but might be nice to tidy up?).

Here's a screenshot of a search block I added using the change in https://github.com/WordPress/gutenberg/pull/31585 where I only updated two of the border radii and left the other fields empty:

![image](https://user-images.githubusercontent.com/14988353/123745981-597e4780-d8f4-11eb-8793-711d9602c167.png)

Reply:
Thanks. This has been fixed up as part of https://github.com/WordPress/gutenberg/pull/33023/commits/dc94737a52ff840b13e34a0087be2476d7c017dc. 

Only the required radius styles are shown for me after this latest commit.